version: "3.9"
services:
  keycloak:
    image: quay.io/keycloak/keycloak:20.0.1
    environment:
      KEYCLOAK_ADMIN: admin
      KEYCLOAK_ADMIN_PASSWORD: admin
    volumes:
      - ./infrastructure/keycloak:/opt/keycloak/data/
    command: start-dev
    ports:
      - "8180:8080"
    networks:
      - kruger-application
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://localhost:8180" ]
      interval: 30s
      timeout: 10s
      retries: 5

  postgres:
    image: postgres:alpine3.17
    environment:
      POSTGRES_USER: admin
      POSTGRES_PASSWORD: admin
      PGDATA: /data/postgres
    volumes:
      - ./infrastructure/postgres:/data/postgres
    ports:
      - "5432:5432"
    networks:
      - kruger-application

  pgadmin:
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@admin.com
      PGADMIN_DEFAULT_PASSWORD: admin
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
      - ./infrastructure/pgadmin:/var/lib/pgadmin
    ports:
      - "5050:80"
    networks:
      - kruger-application

  vm-microservices:
    build: .
    ports:
      - "8080:8080"
    networks:
      - kruger-application
    depends_on:
      keycloak:
        condition: service_healthy


networks:
  kruger-application:
    driver: bridge
